src/components/panel/panel.js

```diff
@@ -1282,18 +1288,24 @@ MdPanelRef.prototype._animateOpen = function() {
  * @private
  */
 MdPanelRef.prototype._animateClose = function() {
+  var onRemoving = this._config['onRemoving'];
+
   var animationConfig = this._config['animation'];
   if (!animationConfig) {
     this.removeClass('md-panel-is-showing');
     this.removeClass('_md-panel-shown');
+    this._$q.when(this._config['scope'].$eval(onRemoving))
     return this._$q.resolve();
   }
 
   var self = this;
   return this._$q(function(resolve, reject) {
-    animationConfig.animateClose(self._$q).then(function(){
+    animationConfig.animateClose(self._$q).then(function() {
       self.removeClass('md-panel-is-showing');
-      resolve(self);
+      onRemoving = self._$q.when(this._config['scope'].$eval(onRemoving))
```

Comment:
Why are you setting the promise result back to onRemoving? Also this is the wrong spot to be calling onRemoving. The panel will still be visible here depending on the animation. This should be in the hide section after the HIDDEN class has been added.